frontend/src/components/Calendar.tsx

```diff
@@ -65,18 +65,36 @@ function Calendar({
       : true;
   };
 
-  const isTimeInAllowedRange = (time: Date) => {
-    if (isTimeInPast(time)) {
+  const timeIsConsecutive = (start: Date, end: Date) => {
+    const consecutive = calendarRef.current?.getApi().getEvents().some(
+      (e) => e.groupId !== 'reservations'
+      && e.start && e.end
+      && ((start.getTime() !== e.start.getTime()) && (end.getTime() !== e.end.getTime()))
+      && (e.start.getTime() === start.getTime()
+        || e.start.getTime() === end.getTime()
+        || e.end.getTime() === start.getTime()
+        || e.end.getTime() === end.getTime()),
+    );
+    return consecutive;
+  };
+
+  const isTimeAllowed = (start: Date, end: Date) => {
+    if (isTimeInPast(start)) {
       calendarRef.current?.getApi().unselect();
       addNewAlert('Aikaa ei voi lisätä menneisyyteen', 'warning');
       return false;
     }
     // TODO: Get timeAtMostInFuture from airfield
-    if (checkIfTimeInFuture && !isTimeAtMostInFuture(time, 7)) {
+    if (checkIfTimeInFuture && !isTimeAtMostInFuture(start, 7)) {
       calendarRef.current?.getApi().unselect();
       addNewAlert('Aikaa ei voi lisätä yli 7 päivän päähän', 'warning');
       return false;
     }
+    if (timeIsConsecutive(start, end)) {
```

Comment:
Tän rajotuksen pitäis koskea vain varausikkunoita. Varauksia pitäis pystyy  tekemään peräkkäin.

Reply:
Jep, nyt se toimii vain varausikkunoille.